lib/src/rewrite.rs

```diff
@@ -842,6 +842,185 @@ pub fn move_commits(
+
+            let connected_target_commits: Vec<_> =
+                RevsetExpression::commits(target_commit_ids.iter().cloned().collect_vec())
+                    .connected()
+                    .evaluate_programmatic(mut_repo)
+                    .map_err(|err| err.expect_backend_error())?
+                    .iter()
+                    .commits(mut_repo.store())
+                    .try_collect()
+                    // TODO: Return evaluation error to caller
+                    .map_err(|err| err.expect_backend_error())?;
+
+            // Commits in the target set should only have other commits in the set as
+            // parents, except the roots of the set, which persist their original
+            // parents.
+            // If a commit in the target set has a parent which is not in the set, but has
+            // an ancestor which is in the set, then the commit will have that ancestor
+            // as a parent instead.
+            let target_commits_internal_parents = {
+                let mut target_commits_internal_parents =
+                    compute_internal_parents_within(&target_commit_ids, &connected_target_commits);
+                target_commits_internal_parents.retain(|id, _| target_commit_ids.contains(id));
+                target_commits_internal_parents
+            };
+
+            // Compute the roots of `target_commits`.
+            let target_root_ids: HashSet<_> = target_commits_internal_parents
+                .iter()
+                .filter(|(_, parents)| parents.is_empty())
+                .map(|(commit_id, _)| commit_id.clone())
+                .collect();
+
+            // Compute the heads of the target set, which will be used as the parents of
+            // the children commits.
+            let target_head_ids = if !children_commit_ids.is_empty() {
+                compute_commits_heads(&target_commit_ids, &connected_target_commits)
+            } else {
+                vec![]
+            };
+
+            // Topological order ensures that any parents of the original commit are
+            // either not in `target_commits` or were already duplicated.
+            for original_commit_id in target_commits.iter().rev() {
+                let original_commit = mut_repo.store().get_commit(original_commit_id)?;
+                let new_parent_ids = if target_root_ids.contains(original_commit_id) {
+                    parent_commit_ids.clone()
+                } else {
+                    target_commits_internal_parents
+                        .get(original_commit_id)
+                        .unwrap()
+                        .iter()
+                        // Replace parent IDs with their new IDs if they were duplicated.
+                        .map(|id| {
+                            duplicated_old_to_new
+                                .get(id)
+                                .map_or_else(|| id.clone(), |commit| commit.id().clone())
```

Comment:
nit: This looks similar to lines 901-902. Write it the same way for consistency?